internals/overlord/servstate/manager.go

```diff
@@ -416,18 +256,14 @@ func (m *ServiceManager) ServiceLogs(services []string, last int) (map[string]se
 // Replan returns a list of services to stop and services to start because
 // their plans had changed between when they started and this call.
 func (m *ServiceManager) Replan() ([]string, []string, error) {
-	mgrPlan, err := m.Plan()
-	if err != nil {
-		return nil, nil, err
-	}
-
 	m.servicesLock.Lock()
 	defer m.servicesLock.Unlock()
 
+	planSnapshot := m.plan()
```

Comment:
Similar here -- let's move this back up so the plan lock is not held at the same time as the services lock.

Reply:
No problem. I think you have a good point, thank you. Reverted in both places.

Reply:
Resolving.